### PR TITLE
feat: add `values` method to `array/fixed-endian-factory`

### LINKED ISSUE #3161: [RFC]: add `values` method to `array/fixed-endian-factory`
### Description

This RFC proposes adding support for a `values` method to [`@stdlib/array/fixed-endian-factory`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/array/fixed-endian-factory).

This method is the equivalent of the [`TypedArray.prototype.values`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/values) method, but for fixed-endian typed arrays.

When adding support, the following tasks should be completed:

1. The `./lib/main.js` file should be updated with the added method.
2. A new test file should be created which includes unit tests testing against expected behavior.
3. Two benchmark files should be added: one measuring baseline performance and another measuring performance as the array length increases.
4. The README should be updated to include documentation for the added method.

To provide a concrete example of what a PR adding the desired method should contain, see https://github.com/stdlib-js/stdlib/commit/956a4624c788689b1bca285856b987ea3aa32eb6, which is a commit adding `forEach` support. This should provide an idea of what is expected.

For an example of specialized typed arrays having customized methods, see

- [`@stdlib/array/bool`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/array/bool)
- [`@stdlib/array/complex128`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/array/complex128)
- [`@stdlib/array/complex64`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/array/complex64)

While the method proposed in this RFC will likely **not** match the implementations found in the above packages, those packages should provide a conceptual idea of what is desired. Do **not** simply copy-paste the code found in those packages.

### Notes

- If you are interested in contributing a PR which addresses this RFC and are a **first-time contributor** or not yet familiar with our project conventions, please do **not** submit LLM-generated code. Please consult our [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md) and the associated [development guide](https://github.com/stdlib-js/stdlib/blob/develop/docs/development.md). Failure to respect project conventions will result in your PR being rejected without review. Thank you for understanding!

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `[RFC]:`.

### PR DESCRIPTION
**Pull request by yuvimittal**
The `values()` method was added to the typed array prototype to allow iteration over the array's values. This method returns an iterator, which can be used in a `for...of` loop or with `next()` calls to access each value in the array sequentially. This change simplifies the process of iterating over typed arrays, ensuring that code interacting with them is more readable and easier to maintain.

Fixes: #3161 

Resolves #3161

## Description

> What is the purpose of this pull request?

This pull request:

add values method to @stdlib/array/fixed-endian-factory

## Related Issues

> Does this pull request have any related issues? 

This pull request:

-   resolves #3161 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by Vinit-Pandit** [comment]
Please change the PR title to 
feat: add `values` method to `array/fixed-endian-factory`

File: lib/node_modules/@stdlib/array/fixed-endian-factory/benchmark/benchmark.values.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	var arr;
+	var i;
+
+	arr = new Float64ArrayFE( 'little-endian', [ 1.0, 2.0, 2.0, 1.0 ] );
+	iter = arr.values();
+
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		iter = arr.values();
+
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/array/fixed-endian-factory/benchmark/benchmark.values.js]:
```suggestion
```

File: lib/node_modules/@stdlib/array/fixed-endian-factory/lib/main.js [hunk 2]
```diff
@@ -850,6 +850,51 @@ function factory( dtype ) { // eslint-disable-line max-lines-per-function, stdli
...
+ * Returns an iterator for array elements.
+ *
+ * @private
+ * @name values
+ * @memberof TypedArray.prototype
+ * @type {Function}
+ * @throws {TypeError} `this` must be a typed array instance
+ * @returns {Object} iterator for array elements
+ */
+	setReadOnly(TypedArray.prototype, 'values', function values() {
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/array/fixed-endian-factory/lib/main.js]:
```suggestion
	setReadOnly( TypedArray.prototype, 'values', function values() {
```

File: lib/node_modules/@stdlib/array/fixed-endian-factory/lib/main.js [hunk 3]
```diff
@@ -850,6 +850,51 @@ function factory( dtype ) { // eslint-disable-line max-lines-per-function, stdli
...
+		var iterator;
+		var index = 0;
+		var value;
+		var isLE = this._isLE;
+		var buf = this._buffer;
+		var len = this._length;
+
+		if (!isTypedArray(this)) {
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/array/fixed-endian-factory/lib/main.js]:
```suggestion
		if ( !isTypedArray( this ) ) {
```

File: lib/node_modules/@stdlib/array/fixed-endian-factory/lib/main.js [hunk 4]
```diff
@@ -850,6 +850,51 @@ function factory( dtype ) { // eslint-disable-line max-lines-per-function, stdli
...
+			}
+		};
+
+		iterator[Symbol.iterator] = function iterator() {
+			return this;
+		};
+
+		return iterator;
+	});
+
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/array/fixed-endian-factory/lib/main.js]: Please add the spaces under the every parans.. same way i have suggested

File: lib/node_modules/@stdlib/array/fixed-endian-factory/test/test.values.js [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof factory, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the `values` method returns an iterator function', function test( t ) {
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/array/fixed-endian-factory/test/test.values.js:37]: Test case like [this](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/array/fixed-endian-factory/test/test.every.js#L43) is missing

File: lib/node_modules/@stdlib/array/fixed-endian-factory/test/test.values.js [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+
+	t.strictEqual( isFunction( iterator.next ), true, 'returns expected iterator' );
+	t.end();
+});
+
+tape( 'the `values` iterator stops when all elements are iterated', function test( t ) {
+	var iterator;
+	var result;
+	var ctor = factory( 'float64' );
+	var arr = new ctor( 'little-endian', [1.0, 2.0, 3.0, 4.0, 5.0] );
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/array/fixed-endian-factory/test/test.values.js]:
```suggestion
	var arr = new ctor( 'little-endian', [ 1.0, 2.0, 3.0, 4.0, 5.0 ] );
```

File: lib/node_modules/@stdlib/array/fixed-endian-factory/test/test.values.js [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+	}, 'returns expected result' );
+
+	t.end();
+});
+
+tape( 'the `values` iterator returns the correct value for typed arrays with different data types', function test( t ) {
+	var iterator;
+	var result;
+	var ctor = factory( 'float64' );
+	var arr = new ctor( 'little-endian', [10.5, 20.5, 30.5] );
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/array/fixed-endian-factory/test/test.values.js]: missing spaces in square braces [ ... ]

File: lib/node_modules/@stdlib/array/fixed-endian-factory/test/test.values.js [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+	}, 'iteration is done' );
+
+	t.end();
+});
+
+tape( 'the `values` iterator works with `for...of` loop', function test( t ) {
+	var values = [];
+	var value;
+	var ctor = factory( 'float64' );
+	var arr = new ctor( 'little-endian', [1.0, 2.0, 3.0, 4.0, 5.0] );
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/array/fixed-endian-factory/test/test.values.js]: missing spaces in square braces [ ... ]

File: lib/node_modules/@stdlib/array/fixed-endian-factory/test/test.values.js [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+
+	for ( value of arr.values() ) {
+		values.push( value );
+	}
+
+	t.deepEqual( values, [1.0, 2.0, 3.0, 4.0, 5.0], 'correctly iterates over values' );
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/array/fixed-endian-factory/test/test.values.js]: missing spaces in square braces [ ... ] . 

please do same for all

**Comment by yuvimittal** [comment]
@MeastroZI  Thanks for the review , i have made the required changes.

**Pull request closed without merging**